source/common/router/router.cc

```diff
@@ -818,21 +813,33 @@ void Filter::UpstreamRequest::decode100ContinueHeaders(Http::HeaderMapPtr&& head
 }
 
 void Filter::UpstreamRequest::decodeHeaders(Http::HeaderMapPtr&& headers, bool end_stream) {
+  // TODO(rodaine): This is actually measuring after the headers are parsed and not the first byte.
```

Comment:
When you fix this for 100-continue etc. please fix the sending-to-downstream side as well.